### ISSUE TITLE
Bug: weird shotgun behaviour both in demo viewer and videorender

### ISSUE THREAD

**Issue by gerdoe-jr** [ddnet/ddnet#8895]
Reported by @nRx on Discord:

https://github.com/user-attachments/assets/a54d21db-8ec4-409b-8320-638f1ae07273

**Comment by ChillerDragon** [comment]
This got fixed in #9174

**Comment by ChillerDragon** [comment]
@KebsCS are you planning to work on this? I feel like you know really well how tunes work right now. If not I will give this a go but I would prefer not to.

**Comment by KebsCS** [comment]
> [@KebsCS](https://github.com/KebsCS) are you planning to work on this? I feel like you know really well how tunes work right now. If not I will give this a go but I would prefer not to.

I don't know how to properly check if server is vanilla. The other workaround would be to throw tunings received from `NETMSGTYPE_SV_TUNEPARAMS` into tuning list, but from my testing this doesn't work well in demos and it still can be wrong when there is no local player. You can give it a go

**Comment by ChillerDragon** [comment]
Imo every server should default to vanilla tunings and then update it to whatever tunings the server sends. All ddnet servers already send ddrace tunings.

**Comment by KebsCS** [comment]
> Imo every server should default to vanilla tunings and then update it to whatever tunings the server sends. All ddnet servers already send ddrace tunings.

Yes, but the problem is in implementation of sending the tunings. The tunes are only received if you're exactly in the same zone, see https://github.com/ddnet/ddnet/pull/9902#issuecomment-2744291758
For demos, the client sends `NETMSGTYPE_SV_TUNEPARAMS` with `MSGFLAG_RECORD | MSGFLAG_NOSEND` but I couldn't properly check where the player is standing to put it into world list, it was always off by few ticks when entering a new zone. Also this doesn't work if there is no recording player's tee or we didn't enter all zones on the map
